src/NHibernate/NHibernateUtil.cs

```diff
@@ -382,10 +382,15 @@ public static void Initialize(object proxy)
 			{
 				((INHibernateProxy)proxy).HibernateLazyInitializer.Initialize();
 			}
-			else if (proxy is IPersistentCollection coll)
+			else if (proxy is ILazyInitializedCollection coll)
 			{
 				coll.ForceInitialization();
 			}
+			else if (proxy is IPersistentCollection persistent)
```

Comment:
Another todo here would be great, like `6.0 TODO: remove once IPersistentCollection derives from ILazyInitializedCollection`.